integration/discovery/discovery_test.go

```diff
@@ -114,580 +77,713 @@ var _ = Describe("DiscoveryService", func() {
-		submitSnapshotRequest(network, "testchannel", 0, org1Peer0, "Snapshot request submitted successfully")
-
-		By("verifying snapshot completed on org1Peer0")
-		verifyNoPendingSnapshotRequest(network, org1Peer0, "testchannel")
-		snapshotDir := filepath.Join(network.PeerDir(org1Peer0), "filesystem", "snapshots", "completed", "testchannel", strconv.Itoa(blockNum))
-
-		By("adding peer org3Peer0 to the network")
-		org3Peer0 = &nwo.Peer{
-			Name:         "peer0",
-			Organization: "Org3",
-			Channels: []*nwo.PeerChannel{
-				{Name: "testchannel", Anchor: true},
-			},
-		}
-		network.Peers = append(network.Peers, org3Peer0)
+	Describe("basic etcdraft network with anchor peers", func() {
+		BeforeEach(func() {
+			By("generating a network with anchor peers defined and adding a 3rd org")
+			// add org3 with one peer (to generate cryptogen and configtx files)
+			config.Organizations = append(config.Organizations, &nwo.Organization{
+				Name:          "Org3",
+				MSPID:         "Org3MSP",
+				Domain:        "org3.example.com",
+				EnableNodeOUs: true,
+				Users:         2,
+				CA:            &nwo.CA{Hostname: "ca"},
+			})
+			config.Consortiums[0].Organizations = append(config.Consortiums[0].Organizations, "Org3")
+			config.Profiles[0].Organizations = append(config.Profiles[0].Organizations, "Org3")
+			config.Peers = append(config.Peers,
+				&nwo.Peer{
+					Name:         "peer0",
+					Organization: "Org3",
+					Channels: []*nwo.PeerChannel{
+						{Name: "testchannel", Anchor: true},
+					},
+				},
+			)
+
+			network = nwo.New(config, testDir, client, StartPort(), components)
+			network.GenerateConfigTree()
+			network.Bootstrap()
+
+			// initially remove Org3 peer0 from the network and later add it back to test joinbysnapshot
+			peers := []*nwo.Peer{}
+			for _, p := range network.Peers {
+				if p.Organization != "Org3" {
+					peers = append(peers, p)
+				}
+			}
+			network.Peers = peers
+
+			By("asserting the genesis block has anchor peers defined")
+			assertAnchorPeers(network, true, "testchannel", "Org1", "Org2", "Org3")
+
+			// Start all the fabric processes
+			var peerProcess ifrit.Process
+			ordererRunner, ordererProcess, peerProcess = network.StartSingleOrdererNetwork("orderer")
+			peerProcesses = append(peerProcesses, peerProcess)
+
+			orderer = network.Orderer("orderer")
+			channelparticipation.JoinOrdererJoinPeersAppChannel(network, "testchannel", orderer, ordererRunner)
+
+			org1Peer0 = network.Peer("Org1", "peer0")
+			org2Peer0 = network.Peer("Org2", "peer0")
+		})
+
+		It("discovers network configuration, endorsers, and peer membership", func() {
+			//
+			// bootstrapping a peer from snapshot
+			//
+			By("generating a snapshot at current block number on org1Peer0")
+			blockNum := nwo.GetLedgerHeight(network, org1Peer0, "testchannel") - 1
+			submitSnapshotRequest(network, "testchannel", 0, org1Peer0, "Snapshot request submitted successfully")
+
+			By("verifying snapshot completed on org1Peer0")
```

Comment:
Not sure this is really needed unless this somehow blocks until the previous snapshot ends, but I guess it's not worth another iteration to remove it